PORTAL/jobs/_utils.py

```diff
@@ -1507,7 +1513,10 @@ def _git(cmd, args, cwd, cfg):
     return text
 
 
-def _git_raw(cmd, args, cwd, cfg, *, GIT=shutil.which('git')):
+_git_path = shutil.which('git')
+
+
+def _git_raw(cmd, args, cwd, cfg, *, GIT=_git_path):
```

Comment:
Why pull out `_git_path`?

Reply:
It's a flake8 rule that discourages function calls in kwargs defaults, since they are evaluated at load time, which can be confusing.

Reply:
I guess flake8 has to play to the lowest common denominator. :)  The keyword-only default was there to make it clear that the constant only applies to that function (and to do so with brevity).  If flake8 doesn't like this (and we don't `noqa: XYZ` that line) then there's no need for the keyword arg.

```suggestion
GIT = shutil.which('git')


def _git_raw(cmd, args, cwd, cfg):
```

or (to keep with the spirit of the keyword arg usage):

```suggestion
def _git_raw(cmd, args, cwd, cfg):
    global GIT
    if 'GIT' not in globals():
        GIT = shutil.which('git')
```

Personally I'd rather `noqa` that line.

Reply:
I just disabled this lint (B008) globally in `setup.cfg`.  It doesn't seem super pertinent.  (The primary reason I like flake8 is for undefined symbols which usually do represent real bugs in uncovered code -- the rest just comes along for the ride, so I'm happy to just suppress in those cases).